src/components/KidsCard.js

```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components';
+import IconButton from './IconButton.js';
+import DeleteIcon from './icons/DeleteIcon.js';
+
+export default function KidsCard({ name, birthDate, kidsId, kidsData, setKidsData }) {
+  function handleDelteKidClick(event) {
+    event.stopPropagation();
+    setKidsData(kidsData.filter(item => item.id !== kidsId));
+  }
```

Comment:
It is considered "bad" practice to pass the setter function of a state to a component. This might lead to bugs when updating the state. I would recommend passing a callback function instead. 
```suggestion
export default function KidsCard({ name, birthDate, kidsId, onDeleteKid }) {
  function handleDelteKidClick(event) {
    event.stopPropagation();
    onDeleteKid(kidsId);
  }
```
The filtering would be done where the state is, in your `App.js`. And you also wouldn't need to passe `kidsData` as props.

Reply:
Thanks a lot I´ve just changed it